BlobStorage/Advanced.cs

```diff
@@ -1584,80 +1332,67 @@ private static async Task CopyBlockBlobAsync(CloudBlobContainer container)
         /// Creates a large block blob, and copies it to a new blob in the same container. If the copy operation
         /// does not complete within the specified interval, abort the copy operation.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <param name="sizeInMb">The size of block blob to create, in MB.</param>
         /// <returns>A Task object.</returns>
-        private static async Task CopyLargeBlockBlobAsync(CloudBlobContainer container, int sizeInMb)
+        private static async Task CopyLargeBlockBlobAsync(BlobContainerClient container, int sizeInMb)
         {
             // Create an array of random bytes, of the specified size.
             byte[] bytes = new byte[sizeInMb * 1024 * 1024];
             Random rng = new Random();
             rng.NextBytes(bytes);
 
             // Get a reference to a new block blob.
-            CloudBlockBlob sourceBlob = container.GetBlockBlobReference("LargeSourceBlob");
+            BlobClient sourceBlob = container.GetBlobClient("LargeSourceBlob");
 
             // Get a reference to the destination blob (in this case, a new blob).
-            CloudBlockBlob destBlob = container.GetBlockBlobReference("copy of " + sourceBlob.Name);
+            BlobClient destBlob = container.GetBlobClient("copy of " + sourceBlob.Name);
 
-            MemoryStream msWrite = null;
-            string copyId = null;
             string leaseId = null;
 
             try
             {
                 // Create a new block blob comprised of random bytes to use as the source of the copy operation.
-                msWrite = new MemoryStream(bytes);
-                msWrite.Position = 0;
-                using (msWrite)
-                {
-                    await sourceBlob.UploadFromStreamAsync(msWrite);
-                }
+                await sourceBlob.UploadAsync(BinaryData.FromString(bytes.ToString()));
 
                 // Lease the source blob for the copy operation to prevent another client from modifying it.
                 // Specifying null for the lease interval creates an infinite lease.
-                leaseId = await sourceBlob.AcquireLeaseAsync(null);
+                var leaseClient = sourceBlob.GetBlobLeaseClient();
+                var blobLease = await leaseClient.AcquireAsync(TimeSpan.FromSeconds(15));
+                leaseId = blobLease.Value.LeaseId;
```

Comment:
we never use this lease id in our operation. This code should be failing due to all the operations not actually supplying a lease. We should use or delete it.